// react-form-wizard-component/dist/react-form-wizard-component.es.js
/*
  * react-form-wizard-component
  * Creator:Parsa Jiravand
  * A modern, accessible React form wizard component with validation, progress tracking, and extensive customization options. Supports React 19, zero dependencies, TypeScript-first, mobile-responsive.
  * v1.1.1
  * MIT License
  */
 
import y, { useState as re, useEffect as L, useMemo as te, useRef as le, useImperativeHandle as me } from "react";
var ne = { exports: {} }, K = {};
/**
 * @license React
 * react-jsx-runtime.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var be;
function xe() {
  if (be) return K;
  be = 1;
  var n = Symbol.for("react.transitional.element"), v = Symbol.for("react.fragment");
  function h(E, u, i) {
    var p = null;
    if (i !== void 0 && (p = "" + i), u.key !== void 0 && (p = "" + u.key), "key" in u) {
      i = {};
      for (var _ in u)
        _ !== "key" && (i[_] = u[_]);
    } else i = u;
    return u = i.ref, {
      $$typeof: n,
      type: E,
      key: p,
      ref: u !== void 0 ? u : null,
      props: i
    };
  }
  return K.Fragment = v, K.jsx = h, K.jsxs = h, K;
}
var C = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var pe;
function ge() {
  return pe || (pe = 1, process.env.NODE_ENV !== "production" && function() {
    function n(e) {
      if (e == null) return null;
      if (typeof e == "function")
        return e.$$typeof === w ? null : e.displayName || e.name || null;
      if (typeof e == "string") return e;
      switch (e) {
        case x:
          return "Fragment";
        case U:
          return "Profiler";
        case q:
          return "StrictMode";
        case H:
          return "Suspense";
        case Z:
          return "SuspenseList";
        case Q:
          return "Activity";
      }
      if (typeof e == "object")
        switch (typeof e.tag == "number" && console.error(
          "Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."
        ), e.$$typeof) {
          case N:
            return "Portal";
          case X:
            return e.displayName || "Context";
          case $:
            return (e._context.displayName || "Context") + ".Consumer";
          case a:
            var t = e.render;
            return e = e.displayName, e || (e = t.displayName || t.name || "", e = e !== "" ? "ForwardRef(" + e + ")" : "ForwardRef"), e;
          case T:
            return t = e.displayName || null, t !== null ? t : n(e.type) || "Memo";
          case O:
            t = e._payload, e = e._init;
            try {
              return n(e(t));
            } catch {
            }
        }
      return null;
    }
    function v(e) {
      return "" + e;
    }
    function h(e) {
      try {
        v(e);
        var t = !1;
      } catch {
        t = !0;
      }
      if (t) {
        t = console;
        var f = t.error, b = typeof Symbol == "function" && Symbol.toStringTag && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return f.call(
          t,
          "The provided key is an unsupported type %s. This value must be coerced to a string before using it here.",
          b
        ), v(e);
      }
    }
    function E(e) {
      if (e === x) return "<>";
      if (typeof e == "object" && e !== null && e.$$typeof === O)
        return "<...>";
      try {
        var t = n(e);
        return t ? "<" + t + ">" : "<...>";
      } catch {
        return "<...>";
      }
    }
    function u() {
      var e = B.A;
      return e === null ? null : e.getOwner();
    }
    function i() {
      return Error("react-stack-top-frame");
    }
    function p(e) {
      if (z.call(e, "key")) {
        var t = Object.getOwnPropertyDescriptor(e, "key").get;
        if (t && t.isReactWarning) return !1;
      }
      return e.key !== void 0;
    }
    function _(e, t) {
      function f() {
        M || (M = !0, console.error(
          "%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://react.dev/link/special-props)",
          t
        ));
      }
      f.isReactWarning = !0, Object.defineProperty(e, "key", {
        get: f,
        configurable: !0
      });
    }
    function Y() {
      var e = n(this.type);
      return D[e] || (D[e] = !0, console.error(
        "Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release."
      )), e = this.props.ref, e !== void 0 ? e : null;
    }
    function j(e, t, f, b, I, W) {
      var d = f.ref;
      return e = {
        $$typeof: R,
        type: e,
        key: t,
        props: f,
        _owner: b
      }, (d !== void 0 ? d : null) !== null ? Object.defineProperty(e, "ref", {
        enumerable: !1,
        get: Y
      }) : Object.defineProperty(e, "ref", { enumerable: !1, value: null }), e._store = {}, Object.defineProperty(e._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: 0
      }), Object.defineProperty(e, "_debugInfo", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: null
      }), Object.defineProperty(e, "_debugStack", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: I
      }), Object.defineProperty(e, "_debugTask", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: W
      }), Object.freeze && (Object.freeze(e.props), Object.freeze(e)), e;
    }
    function F(e, t, f, b, I, W) {
      var d = t.children;
      if (d !== void 0)
        if (b)
          if (P(d)) {
            for (b = 0; b < d.length; b++)
              k(d[b]);
            Object.freeze && Object.freeze(d);
          } else
            console.error(
              "React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead."
            );
        else k(d);
      if (z.call(t, "key")) {
        d = n(e);
        var m = Object.keys(t).filter(function(ee) {
          return ee !== "key";
        });
        b = 0 < m.length ? "{key: someKey, " + m.join(": ..., ") + ": ...}" : "{key: someKey}", c[d + b] || (m = 0 < m.length ? "{" + m.join(": ..., ") + ": ...}" : "{}", console.error(
          `A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`,
          b,
          d,
          m,
          d
        ), c[d + b] = !0);
      }
      if (d = null, f !== void 0 && (h(f), d = "" + f), p(t) && (h(t.key), d = "" + t.key), "key" in t) {
        f = {};
        for (var A in t)
          A !== "key" && (f[A] = t[A]);
      } else f = t;
      return d && _(
        f,
        typeof e == "function" ? e.displayName || e.name || "Unknown" : e
      ), j(
        e,
        d,
        f,
        u(),
        I,
        W
      );
    }
    function k(e) {
      G(e) ? e._store && (e._store.validated = 1) : typeof e == "object" && e !== null && e.$$typeof === O && (e._payload.status === "fulfilled" ? G(e._payload.value) && e._payload.value._store && (e._payload.value._store.validated = 1) : e._store && (e._store.validated = 1));
    }
    function G(e) {
      return typeof e == "object" && e !== null && e.$$typeof === R;
    }
    var S = y, R = Symbol.for("react.transitional.element"), N = Symbol.for("react.portal"), x = Symbol.for("react.fragment"), q = Symbol.for("react.strict_mode"), U = Symbol.for("react.profiler"), $ = Symbol.for("react.consumer"), X = Symbol.for("react.context"), a = Symbol.for("react.forward_ref"), H = Symbol.for("react.suspense"), Z = Symbol.for("react.suspense_list"), T = Symbol.for("react.memo"), O = Symbol.for("react.lazy"), Q = Symbol.for("react.activity"), w = Symbol.for("react.client.reference"), B = S.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE, z = Object.prototype.hasOwnProperty, P = Array.isArray, V = console.createTask ? console.createTask : function() {
      return null;
    };
    S = {
      react_stack_bottom_frame: function(e) {
        return e();
      }
    };
    var M, D = {}, l = S.react_stack_bottom_frame.bind(
      S,
      i
    )(), J = V(E(i)), c = {};
    C.Fragment = x, C.jsx = function(e, t, f) {
      var b = 1e4 > B.recentlyCreatedOwnerStacks++;
      return F(
        e,
        t,
        f,
        !1,
        b ? Error("react-stack-top-frame") : l,
        b ? V(E(e)) : J
      );
    }, C.jsxs = function(e, t, f) {
      var b = 1e4 > B.recentlyCreatedOwnerStacks++;
      return F(
        e,
        t,
        f,
        !0,
        b ? Error("react-stack-top-frame") : l,
        b ? V(E(e)) : J
      );
    };
  }()), C;
}
var he;
function je() {
  return he || (he = 1, process.env.NODE_ENV === "production" ? ne.exports = xe() : ne.exports = ge()), ne.exports;
}
var o = je();
const Se = y.memo(y.forwardRef(
  ({
    id: n,
    title: v,
    icon: h,
    shape: E,
    color: u = "#2196f3",
    isActive: i,
    index: p,
    currentStep: _,
    isVisible: Y = !0,
    isDisabled: j = !1,
    hasValidationError: F = !1,
    showProgressBar: k,
    layout: G,
    inlineStep: S = !1,
    darkColor: R,
    darkIconColor: N,
    removeBackgroundTab: x,
    removeBackgroundTabTransparentColor: q,
    showErrorOnTab: U,
    showErrorOnTabColor: $ = "red",
    onClick: X
  }, a) => {
    const H = i ? "active" : "", Z = j ? "not-allowed" : E === "square" ? "default" : "pointer", [T, O] = y.useState(!1);
    y.useEffect(() => {
      i && O(!0);
    }, [i]);
    const Q = () => {
      const P = {
        border: "2px solid " + u
      };
      return R && (P.border = "2px solid " + R), (U || F) && (P.border = "2px solid " + $), G === "vertical" ? {
        ...P,
        rotate: "90deg",
        animation: "slideInVertical 0.3s forwards"
      } : P;
    }, w = () => {
      if (i && N)
        return { color: N || u };
      if (i && T)
        return { color: "white" };
      if (T && N)
        return { color: N || u };
      if (T)
        return { color: "white" };
    }, B = () => (U || F) && T && p <= _ ? $ : T && !x ? R || u : "";
    y.useImperativeHandle(a, () => ({
      setChecked: (P) => {
        O(P);
      }
    }));
    const z = () => h ? typeof h == "string" ? /* @__PURE__ */ o.jsx("i", { className: h, style: w() }) : h : /* @__PURE__ */ o.jsx("span", { style: w(), children: p + 1 });
    return Y ? /* @__PURE__ */ o.jsxs(
      "li",
      {
        className: `${H}`,
        style: {
          position: "relative"
        },
        children: [
          k && T && p <= _ && /* @__PURE__ */ o.jsx(
            "div",
            {
              className: "smooth-border-left-to-right",
              style: Q()
            }
          ),
          /* @__PURE__ */ o.jsxs(
            "a",
            {
              className: `${i ? "active" : ""} ${S ? "inline-step" : ""}`,
              style: { cursor: Z },
              onClick: j ? void 0 : X,
              role: "tab",
              "aria-selected": i,
              "aria-controls": `${n ?? `step-${p}`}-panel`,
              id: n ?? `step-${p}`,
              tabIndex: i ? 0 : -1,
              "aria-disabled": j,
              children: [
                /* @__PURE__ */ o.jsx(
                  "div",
                  {
                    className: `wizard-icon-circle md ${T ? "checked" : ""} ${E === "square" ? "square_shape" : ""}`,
                    style: {
                      backgroundColor: x ? "transparent" : T ? R || u : "",
                      border: x ? "unset" : ""
                    },
                    children: /* @__PURE__ */ o.jsx(
                      "div",
                      {
                        className: `wizard-icon-container ${E === "square" ? "square_shape" : ""}`,
                        style: {
                          backgroundColor: B()
                        },
                        children: /* @__PURE__ */ o.jsx(
                          "span",
                          {
                            className: "wizard-icon",
                            style: x ? {
                              backgroundColor: q || "white",
                              padding: "10px"
                            } : {},
                            children: z()
                          }
                        )
                      }
                    )
                  }
                ),
                /* @__PURE__ */ o.jsx(
                  "span",
                  {
                    className: `stepTitle ${i ? "active" : ""}`,
                    style: {
                      color: U && T && p <= _ || F && T && p <= _ ? $ : T ? R || u : "",
                      marginTop: S ? "" : "8px",
                      padding: S ? "0 10px" : "0"
                    },
                    children: v
                  }
                )
              ]
            }
          )
        ]
      },
      p
    ) : null;
  }
)), ce = y.memo(({
  darkTextColor: n,
  darkButtonColor: v,
  onClick: h,
  children: E
}) => /* @__PURE__ */ o.jsx(
  "button",
  {
    className: "wizard-btn",
    type: "button",
    style: {
      color: n,
      backgroundColor: v
    },
    onClick: h,
    children: E
  }
)), Ee = (n) => `step-${n + 1}`, Ne = (n, v, h) => {
  if (!(n.condition ? n.condition(h) : !0)) return null;
  const u = n.validate ? n.validate(h) : !0, i = u === !0, p = typeof n.content == "function" ? n.content(h) : n.content;
  return {
    id: n.id ?? Ee(v),
    title: n.title ?? `Step ${v + 1}`,
    icon: n.icon,
    content: p,
    showErrorOnTab: n.showErrorOnTab ?? !i,
    showErrorOnTabColor: n.showErrorOnTabColor ?? "red",
    isValid: i,
    validationError: typeof u == "string" ? () => u : void 0
  };
}, Oe = (n, v, h) => {
  var Y, j;
  const E = n.props.condition;
  if (!(E ? E(h) : !0)) return null;
  const i = (j = (Y = n.props).validate) == null ? void 0 : j.call(Y, h), p = n.props.isValid ?? !0, _ = i === void 0 ? p : i === !0;
  return {
    id: n.props.id ?? Ee(v),
    title: n.props.title ?? `Step ${v + 1}`,
    icon: n.props.icon,
    content: n.props.children,
    showErrorOnTab: n.props.showErrorOnTab ?? !_,
    showErrorOnTabColor: n.props.showErrorOnTabColor ?? "red",
    validationError: n.props.validationError,
    isValid: _
  };
}, Pe = y.forwardRef(
  ({
    title: n,
    subtitle: v = "",
    shape: h = "",
    color: E = "#2196f3",
    children: u,
    schema: i,
    data: p,
    onDataChange: _,
    nextButtonText: Y = "Next",
    nextButtonTemplate: j,
    backButtonText: F = "Back",
    backButtonTemplate: k,
    finishButtonText: G = "Finish",
    finishButtonTemplate: S,
    stepSize: R = "md",
    layout: N = "horizontal",
    startIndex: x = 0,
    disableBackOnClickStep: q = !1,
    showProgressBar: U = !0,
    inlineStep: $ = !1,
    darkMode: X = !1,
    customDarkModeColor: a = {},
    removeBackgroundTab: H = !1,
    removeBackgroundTabTransparentColor: Z = "",
    onComplete: T,
    onTabChange: O
  }, Q) => {
    var fe, de;
    const [w, B] = re(X), [z, P] = re(
      p ?? (i == null ? void 0 : i.initialData) ?? {}
    );
    L(() => {
      B(!!X);
    }, [X]), L(() => {
      p && P(p);
    }, [p]);
    const V = te(
      () => ({
        data: z,
        currentStep: 0,
        stepIndex: 0
      }),
      [z]
    ), M = te(() => i ? i.steps.map(
      (r, s) => Ne(r, s, {
        ...V,
        stepIndex: s
      })
    ).filter((r) => r !== null) : [], [i, V]), D = te(() => u ? y.Children.toArray(
      u
    ).map(
      (s, g) => Oe(s, g, {
        ...V,
        stepIndex: g
      })
    ).filter((s) => s !== null) : [], [u, V]), l = i ? M : D, J = x >= 0 && x < l.length ? x : 0, [c, e] = re(J), [t, f] = re(J);
    L(() => {
      c > t && f(c);
    }, [c, t]), L(() => {
      c >= l.length && e(Math.max(0, l.length - 1));
    }, [c, l.length]);
    const b = le([]);
    L(() => {
      b.current = l.map(
        (r, s) => b.current[s] ?? y.createRef()
      );
    }, [l]), L(() => {
      b.current.forEach((r, s) => {
        var g;
        (g = r == null ? void 0 : r.current) == null || g.setChecked(s <= t);
      });
    }, [t, l.length]);
    const I = te(() => {
      const r = l[c];
      return r ? r.isValid : !1;
    }, [l, c]), W = y.useCallback(() => {
      const r = l[c];
      r != null && r.validationError && typeof r.validationError == "function" && r.validationError();
    }, [l, c]), d = y.useCallback((r, s = !1) => {
      r < 0 || r >= l.length || !s && q && r > c || !s && r > t || e(r);
    }, [l.length, q, c, t]), m = y.useCallback(() => {
      if (!(c >= l.length - 1)) {
        if (!I) {
          W();
          return;
        }
        e((r) => Math.min(r + 1, l.length - 1));
      }
    }, [c, l.length, I, W]), A = y.useCallback(() => {
      e((r) => Math.max(r - 1, 0));
    }, []), ee = y.useCallback(() => {
      if (!I) {
        W();
        return;
      }
      T == null || T(z);
    }, [I, W, T, z]);
    L(() => {
      var s;
      const r = Math.max(0, c - 1);
      O == null || O({
        prevIndex: r,
        nextIndex: c,
        stepId: (s = l[c]) == null ? void 0 : s.id
      });
    }, [c, O, l]), me(Q, () => ({
      nextTab: m,
      prevTab: A,
      reset: () => {
        e(J), f(J);
      },
      activeAll: () => {
        f(Math.max(0, l.length - 1));
      },
      goToTab: (r) => d(r, !0),
      goToTabById: (r) => {
        const s = l.findIndex((g) => g.id === r);
        s !== -1 && d(s, !0);
      },
      setData: (r) => {
        P(r), _ == null || _(r);
      },
      getData: () => z
    })), L(() => {
      const r = (s) => {
        var g;
        (g = document.activeElement) != null && g.closest(".react-form-wizard") && (s.key === "ArrowRight" && (s.preventDefault(), m()), s.key === "ArrowLeft" && (s.preventDefault(), A()), s.key === "Home" && (s.preventDefault(), d(0, !0)), s.key === "End" && (s.preventDefault(), d(l.length - 1, !0)));
      };
      return document.addEventListener("keydown", r), () => document.removeEventListener("keydown", r);
    }, [l.length, m, A, d]);
    const oe = le(0), ue = le(0), ve = (r) => {
      oe.current = r.touches[0].clientX, ue.current = r.touches[0].clientY;
    }, Te = (r) => {
      const s = r.changedTouches[0].clientX, g = r.changedTouches[0].clientY, ie = oe.current - s, Re = ue.current - g;
      Math.abs(ie) > Math.abs(Re) && Math.abs(ie) > 50 && (ie > 0 ? m() : A());
    }, ae = {
      backgroundColor: w && (a != null && a.buttons) ? a.buttons : E,
      borderColor: w && (a != null && a.buttons) ? a.buttons : E,
      color: w && (a != null && a.buttonsText) ? a.buttonsText : "unset",
      borderRadius: "4px"
    }, we = N === "vertical" ? "vertical" : "horizontal", _e = $ ? "inline" : "", ye = $ ? !1 : U, se = l[c];
    return /* @__PURE__ */ o.jsxs(
      "div",
      {
        className: `react-form-wizard ${R} ${we} ${_e}`,
        role: "region",
        "aria-label": "Form Wizard",
        "aria-describedby": "wizard-description",
        onTouchStart: ve,
        onTouchEnd: Te,
        children: [
          /* @__PURE__ */ o.jsxs("div", { className: "wizard-header", children: [
            /* @__PURE__ */ o.jsxs("div", { id: "wizard-description", className: "sr-only", children: [
              "Form wizard with ",
              l.length,
              " steps. Currently on step ",
              c + 1,
              "."
            ] }),
            typeof n == "string" ? /* @__PURE__ */ o.jsxs(o.Fragment, { children: [
              /* @__PURE__ */ o.jsx(
                "h4",
                {
                  className: "wizard-title",
                  style: w && a.title ? { color: a.title } : {},
                  children: n
                }
              ),
              /* @__PURE__ */ o.jsx(
                "p",
                {
                  className: "category",
                  style: w && a.subtitle ? { color: a.subtitle } : {},
                  children: v
                }
              )
            ] }) : n
          ] }),
          /* @__PURE__ */ o.jsxs("div", { className: "wizard-navigation", children: [
            /* @__PURE__ */ o.jsx(
              "ul",
              {
                className: `form-wizard-steps wizard-nav wizard-nav-pills ${h} ${R}`,
                style: { borderColor: E },
                role: "tablist",
                "aria-label": "Form steps",
                children: l.map((r, s) => /* @__PURE__ */ o.jsx(
                  Se,
                  {
                    ref: b.current[s],
                    id: r.id,
                    title: r.title,
                    icon: r.icon,
                    shape: h,
                    color: E,
                    isActive: s === c,
                    index: s,
                    currentStep: c,
                    isVisible: !0,
                    isDisabled: q ? s !== c : s > t,
                    hasValidationError: !!(r.showErrorOnTab && !r.isValid),
                    layout: N,
                    showProgressBar: ye,
                    inlineStep: $,
                    darkColor: w ? a.tab : "",
                    darkIconColor: w ? a.tabIconColor : "",
                    removeBackgroundTab: H,
                    removeBackgroundTabTransparentColor: Z,
                    showErrorOnTab: r.showErrorOnTab,
                    showErrorOnTabColor: r.showErrorOnTabColor,
                    onClick: () => d(s)
                  },
                  r.id
                ))
              }
            ),
            /* @__PURE__ */ o.jsx(
              "div",
              {
                className: "wizard-tab-content",
                role: "tabpanel",
                "aria-labelledby": ((fe = l[c]) == null ? void 0 : fe.id) ?? `step-${c}`,
                id: `${((de = l[c]) == null ? void 0 : de.id) ?? `step-${c}`}-panel`,
                children: se == null ? void 0 : se.content
              }
            )
          ] }),
          /* @__PURE__ */ o.jsxs("div", { className: "wizard-card-footer clearfix", children: [
            c > 0 && (k ? k(A) : /* @__PURE__ */ o.jsx("div", { className: "wizard-footer-left", style: ae, children: /* @__PURE__ */ o.jsx(
              ce,
              {
                darkTextColor: w ? a == null ? void 0 : a.buttonsText : "",
                darkButtonColor: w ? a == null ? void 0 : a.buttons : "",
                onClick: A,
                children: F
              }
            ) })),
            c < l.length - 1 && (j ? j(m) : /* @__PURE__ */ o.jsx("div", { className: "wizard-footer-right", style: ae, children: /* @__PURE__ */ o.jsx(
              ce,
              {
                darkTextColor: w ? a == null ? void 0 : a.buttonsText : "",
                darkButtonColor: w ? a == null ? void 0 : a.buttons : "",
                onClick: m,
                children: Y
              }
            ) })),
            c === l.length - 1 && (S ? S(ee) : /* @__PURE__ */ o.jsx("div", { className: "wizard-footer-right", style: ae, children: /* @__PURE__ */ o.jsx(
              ce,
              {
                darkTextColor: w ? a == null ? void 0 : a.finishButtonText : "",
                darkButtonColor: w ? a == null ? void 0 : a.finishButton : "",
                onClick: ee,
                children: G
              }
            ) }))
          ] })
        ]
      }
    );
  }
), ze = y.memo(Pe), Ae = ({ children: n, isValid: v = !0 }) => /* @__PURE__ */ o.jsx(o.Fragment, { children: v && n }), Ie = Object.assign(ze, { TabContent: Ae });
export {
  Ae as TabContent,
  Ie as default
};
